doc/source/reference.rst

```diff
@@ -0,0 +1,218 @@
+   different cases, and is semantically close to the actions that will happen
+   to deploy OpenStack. `run.sh` itself is just an interface, behind the
+   scenes, it runs a `DEPLOYMENT_MECHANISM` dependant script starting the
+   appropriate ansible playbooks for the step called.
+
+... A shell script for this interface?
+   It was easier to start with a shell script than writing a CLI in <insert
+   language here>, mostly because the shell script grew out of usage and
+   CI needs.
+
+... Installing from sources?
+   Neither the socok8s repo nor the OpenStack-Helm project's repositories
+   have been packaged for Leap/SLE 15.
+
+Image building process
+======================
+
+Upstream process
+----------------
+
+The OpenStack-Helm project tries to be agnostic of the images by
+providing the ability for deployers to override any image used in the
+charts.
+
+However, the OpenStack-Helm project has a repository, `openstack-helm-images`,
+containing a reference implementation for the images. That repository
+holds the images used for the OpenStack-Helm project charts.
+
+These images are built with Docker.
+`openstack-helm-images` provides Dockerfiles for all the non-OpenStack images.
+For the OpenStack images, `openstack-helm-images` only has a shell script, a
+thin wrapper around `LOCI`.
+
+`LOCI` is the official OpenStack project to build OCI compliant images of
+OpenStack. This follows a source based deployment model.
+
+In the future, `openstack-helm-images` could theoretically add images for
+OpenStack which would be based on packages, by simply providing the appropriate
+Dockerfiles.
+
+socok8s process
+---------------
+
+socok8s leverages the existing OSH-images code.
+
+When running the `build_images` step, the `localhost` asks the `deployer` to
+build images based on the code that was checked in on the `deployer` node
+using the `vars/manifest.yml`.
+
+For the non-LOCI images, the `suse-build-images` role invoked in the
+`build_images` step is simply running a `docker build` command.
+
+For the LOCI images, the `suse-build-images` role simply runs the command
+available in `openstack-helm-images` calling the LOCI build.
+
+OpenStack-Helm chart overrides
+==============================
+
+Helm chart values overriding principle
+--------------------------------------
+
+A helm chart installation
+(See https://helm.sh/docs/using_helm/#customizing-the-chart-before-installing )
+accepts an argument named ``--values`` or ``-f``.
+
+This argument expects the filename of a YAML file present on the
+helm client machine. It can be specified multiple times, and
+the rightmost file will take precedence.
+
+OpenStack-Helm scripts
+----------------------
+
+The OpenStack-Helm project provides shell scripts to deploy the helm charts,
+with overrides per context (for example, multinode).
+
+Those shell scripts calling the helm installation include an environment
+variable, to allow users to pass extra arguments.
```

Comment:
Please provide an example of some of those shell scripts

Reply:
Good idea, with an include.